source/fem-dem/cfd_dem_coupling.cc

```diff
@@ -1267,7 +1267,22 @@ CFDDEMSolver<dim>::print_particles_summary()
         << "--------------------------------------------------------------------------"
         << "--------------------------------------------------------------------------"
         << std::endl;
+
+      particle_velocity_table.add_value(
+        "time", this->simulation_control->get_current_time());
+      particle_velocity_table.add_value(
+        "particle-velocity", particle_properties[DEM::PropertiesIndex::v_y]);
     }
+
+  {
+    std::string filename =
+      this->simulation_parameters.simulation_control.output_folder +
+      "particle_velocity" + ".dat";
+    std::ofstream output(filename.c_str());
+    particle_velocity_table.set_precision("time", 12);
+    particle_velocity_table.set_precision("particle-velocity", 12);
+    particle_velocity_table.write_text(output);
```

Comment:
Ohh I see, you wanted to write it to a file?  Then I guess that's ok, but I would add more than the y component of the velocity.
I am not sure about adding another parameter. I'm wondering what is the use of this file other than for a few comparison? Because you can get the particles velocity using the pvd and the vtu files, so what's the point of having them in a text file?

Reply:
I agree. There is no reason to have it in a separate file. If you want, I can help you to do it in the postprocessing using pyvista.

Reply:
True I only had it for the comparison of the sedimentation velocity of different void fraction models. This can be removed entirely. its useless except in this case. I will remove it then.